restlib/restlib.go

```diff
@@ -49,14 +49,16 @@ func unmarshal(resp *http.Response, body []byte, respStruct interface{}) (*HTTPR
 		return makeHTTPResult(resp, body, nil), nil
 	}
 
+	contentType := resp.Header.Get("Content-Type")
+
 	e := reflect.ValueOf(respStruct).Elem()
-	if e.Kind() == reflect.String {
+	kind := e.Kind()
+	if kind == reflect.String || (kind == reflect.Slice && strings.Contains(contentType, "image")) {
```

Comment:
Is it possible to generalise this check to byte arrays? 
```
Current:  Is this a slice and an image.
Proposed: Is this a slice of bytes.
```

Reply:
Good catch, the "text/plain" for ApiDoc is slice as well, I'll have a look if it can further check the type in the slice.